motionplan/inverseKinematics.go

```diff
@@ -2,23 +2,235 @@ package motionplan
 
 import (
 	"context"
+	"math"
+	"runtime"
+	"sort"
+
+	"github.com/edaniels/golog"
+	"go.viam.com/utils"
 
 	"go.viam.com/rdk/referenceframe"
-	spatial "go.viam.com/rdk/spatialmath"
+	"go.viam.com/rdk/spatialmath"
+)
+
+// default values for inverse kinematics.
+const (
+	// If an IK solution scores below this much, return it immediately.
+	defaultMinIkScore = 0.
+
+	// Number of IK solutions that should be generated before stopping.
+	defaultSolutionsToSeed = 50
 )
 
-// InverseKinematics defines an interface which, provided with a goal position and seed inputs, will output all found
-// solutions to the provided channel until cancelled or otherwise completes.
-type InverseKinematics interface {
-	// Solve receives a context, the goal arm position, and current joint angles.
-	Solve(context.Context, chan<- []referenceframe.Input, spatial.Pose, []referenceframe.Input, Metric, int) error
+var defaultNumThreads = runtime.NumCPU() / 2
```

Comment:
Nit: In some places we set defaultNumThreads as NumCPU / 2, others we set it as NumCPU / 4 (with some caveats in places). We should settle on one calculation for this.

Reply:
`Looks like its one other place, in some test code.  I'll change it to be `NumCPU/2`